include/highfive/bits/H5Node_traits_misc.hpp

```diff
@@ -180,6 +197,35 @@ inline bool NodeTraits<Derivate>::exist(const std::string& node_name) const {
     return (val > 0 );
 }
 
+template <typename Derivate>
+inline bool NodeTraits<Derivate>::exist(const std::string& group_path,
+                                        bool recursive) const {
+    if (recursive) {
+        std::string path = group_path;
```

Comment:
You do not really need memory allocation nor string assignment here. You can do the "view" and the recursion entirely with Iterators.